src/serial.rs

```diff
@@ -32,49 +38,62 @@ impl From<panel_protocol::Error> for Error {
     }
 }
 
-pub struct SerialProtocol<PINS> {
+pub struct SerialProtocol<'a> {
     protocol: CommandReader,
-    serial: Serial<stm32::USART1, PINS>,
+    usb_device: UsbDevice<'a, UsbBus<Peripheral>>,
+    usb_serial_device: SerialPort<'a, UsbBus<Peripheral>>,
+    read_buf: [u8; MAX_COMMAND_LEN],
 }
 
-impl<PINS: serial::Pins<stm32f1xx_hal::pac::USART1>> SerialProtocol<PINS> {
+impl<'a> SerialProtocol<'a> {
     pub fn new(
-        usart1: stm32::USART1,
-        usart_pins: PINS,
-        afio: &mut afio::Parts,
-        apb: &mut rcc::APB2,
-        clocks: rcc::Clocks,
+        usb_device: usb_device::device::UsbDevice<
+            'a,
+            stm32f1xx_hal::usb::UsbBus<stm32f1xx_hal::usb::Peripheral>,
```

Comment:
can we make this a custom type just to make this a bit more readable, and/or import `usb_device::device::UsbDevice` and `stm32f1xx_hal::usb::UsbBus` since it feels too verbose up in here